### ISSUE TITLE
Show weeks not days | Plant creation is always "plantation time"

### ISSUE THREAD

**Issue by masterofobzene** [7LPdWcaW/GrowTracker-Android#268]
- It doesn't show the weeks from grow stages, only days. This forces the user to make an unnecessary calculation to convert those days into weeks which are more frequently used as a time measurement in farming than days. 

- When creating a plant, for some reason even if you tell the app that you planted 3 months ago, it will say that you planted it now showing minutes since plantation (the app believes that time of creation = plantation, on all cases).

**Comment by 7LPdWcaW** [comment]
Hi, for your second point, I am unable to recreate this. Setting the planted date when creating a new plant sets the time ago correctly. The "Planted" date is under the hood is a "plant stage" (like germination) and should be the very first entry in the actions history. if that is missing, that may be the cause of the issue.

**Comment by masterofobzene** [comment]
In my case if I create a plant now, and lets say the plant is in flowering stage, I would input the sow time (planted time) back in time 3 months ago, but after completing the new plant profile, the card reads (in main screen) "Planted: 50 min ago Flowering: 34 days". I can see the correct dates on the calendar but I only use the main cards to quickly check age of the plant. I honestly would use the app just for that as it auto-updates the age and is better than manually editing an excel. Don'r really use other features since watering depends on how dry the soil is and can vary, fertilizer configuration is pointless because once you make the schedule you already know how to feed the plant (or not, but you would still be doing it wrong with or without the app configured since you configured the wrong data in there too, its redundant)

~~Also, weeks are more useful than days.~~
 I see you added it to milestone.

**Comment by 7LPdWcaW** [comment]
So the app wasn't designed with retroactively adding in data in mind, but it should still be possible. When creating the plant, set the plant date to the date the seed was sowed, then in the details page, select **Plant stage** and move to the flower and set the date. Then verify the order (from top to bottom with bottom being the **oldest** entry) in the History tab that the "Planted" stage is the last item.

If not, you can long press and hold on the entry and drag it below. The date of the entry must still be in the past otherwise the calculation will not be correct. You can change the date by clicking the 3 dots on the card and edit action.

Glad you're finding good use out of the app still! It was originally created for my own use-case and I think your days/weeks conversion makes sense in retrospect.

HTH, if not, and you dont mind sharing your plant data, you can email me the json file and i'll manually edit it for you (found under `/Android/data/me.anon.grow/files/` or Settings ->export

**Comment by 9YbQiuEohUu1** [comment]
> In my case if I create a plant now, and lets say the plant is in flowering stage, I would input the sow time (planted time) back in time 3 months ago, but after completing the new plant profile, the card reads (in main screen) "Planted: 50 min ago Flowering: 34 days".

Regarding the plant creation issue, I just tried to reproduce it by creating four different plants retroactively planted on April 1, May 1, June 1, and July 1, and they all display correctly, on all three views.

![Screenshot_2022-07-16-08-48-45-321_me anon grow](https://user-images.githubusercontent.com/50482952/179342716-c07e4c3f-2993-4111-84f2-40a80184f67c.jpg)


Note that when creating the new plant, you have two fields to edit - **Plant stage** (under **Stage details**) and **Plant date** (under **Grow details** below it). It's the latter that the app uses as a start date, so make sure to change if you create the plant retroactively, otherwise you indeed end up with time paradoxes.

![Screenshot_2022-07-16-09-14-36-888_me anon grow](https://user-images.githubusercontent.com/50482952/179342678-e07b6a63-4405-4e34-9fe3-55cec2031f30.jpg)

You can also edit **Plant date** at any time if you forget to set the proper date when creating the entry.

**Comment by 9YbQiuEohUu1** [comment]
Now, regarding weeks being more useful than days, I beg to differ. It might be true for some users but ultimately I think it depends on what plants you are growing or even what you are more used to. In my case, I find days far more convenient as a measurement, in part because that's what used throughout the horticultural documentation I use and online discussions I participate in. 

The optimal solution to this would be a setting that would allow choosing the time units. Something like:

Display time as:

- Days
- Weeks/days
- Months/weeks/days

(Maybe then I can try the week-based format and understand how wrong I was to stick with days).

The bottom line is, weeks as an option would be a welcome addition, weeks as the new norm would be an inconvenience to some users.

**Comment by masterofobzene** [comment]
So the owner of the app aknowledges the issue and even says it will change the behaviour of days into weeks but you say nothing of what I say happens.
Good luck.
